### PR TITLE
feat: enable simulation from GeoModel

### PR DESCRIPTION
**Pull request by asalzburger**
This PR adds the necessary infrastructure and Python bindings to run simulation (and `material_recording`) from `GeoModel`.

The Plugin is activated with the CMake configuration
`ACTS_BUILD_PLUGIN_GEOMODEL=ON` 

It mimics exactly how this is done for DD4hep in terms of Detector construction, etc.
It *does not* contain the translation to Gen1/Gen2 reconstruction geometry.

The plot attached shows the material recording output starting from ATLAS' `ITK_Sqlite.db`.

It has been produced with
```sh
python <path_to_your_acts_source>/Examples/Scripts/Python/material_recording.py --input ITK_Sqlite.db
```

ITkG4_ACTS

@noemina - this is the first piece for the summer student project to start from.

### CODE REVIEW

File: Examples/Algorithms/Geant4/src/GeoModelDetectorConstruction.cpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+G4VPhysicalVolume* GeoModelDetectorConstruction::Construct() {
+  if (m_g4World == nullptr) {
+    ExtParameterisedVolumeBuilder builder(m_geoModelTree.worldVolumeName);
+    G4LogicalVolume* g4WorldLog = builder.Build(m_geoModelTree.worldVolume);
+    m_g4World =
+        new G4PVPlacement(nullptr, G4ThreeVector(), g4WorldLog,
+                          m_geoModelTree.worldVolumeName, nullptr, false, 0);
+  }
```
Comment (ntadej) [inline Examples/Algorithms/Geant4/src/GeoModelDetectorConstruction.cpp:42]: I think region creators that I recently added to DD4Hep are missing here. It's basically just a loop over them.

Comment (asalzburger) [reply Examples/Algorithms/Geant4/src/GeoModelDetectorConstruction.cpp:47]: Indeed! Updated.

File: Examples/Python/src/GeoModelStub.cpp [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+namespace Acts {
+namespace Python {
+struct Context;
+}  // namespace Python
+}  // namespace Acts
+
+namespace Acts::Python {
+void addGeoModel(Context& /*ctx*/) {}
+}  // namespace Acts::Python
```
Comment (AJPfleger) [inline Examples/Python/src/GeoModelStub.cpp:17]:
```suggestion
namespace Acts::Python {
struct Context;
void addGeoModel(Context& /*ctx*/) {}
}  // namespace Acts::Python
```

File: Plugins/GeoModel/include/Acts/Plugins/GeoModel/GeoModelReader.hpp [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+/// @brief Read the GeoModel from the database
+///
+/// @param dbPath path to the database
+///
+/// @return world/top volume of the GeoModel tree in memory
+GeoModelTree readFromDb(const std::string& dbPath);
+
+}  // namespace GeoModelReader
+
+}  // namespace Acts
```
Comment (AJPfleger) [inline Plugins/GeoModel/include/Acts/Plugins/GeoModel/GeoModelReader.hpp:30]:
```suggestion
namespace Acts::GeoModelReader {

/// @brief Read the GeoModel from the database
///
/// @param dbPath path to the database
///
/// @return world/top volume of the GeoModel tree in memory
GeoModelTree readFromDb(const std::string& dbPath);

}  // namespace Acts::GeoModelReader
```

but I think clang-tidy should complain about this anyway

File: Plugins/GeoModel/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  ActsPluginGeoModel SHARED
+  src/GeoModelReader.cpp)
+target_include_directories(
+    ActsPluginGeoModel
+  PUBLIC
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
+target_include_directories(
+      ActsPluginGeoModel
+      SYSTEM PUBLIC)
```
Comment (AJPfleger) [inline Plugins/GeoModel/CMakeLists.txt:13]: Does this work without any include directories?

Comment (asalzburger) [reply Plugins/GeoModel/CMakeLists.txt:13]: Removed this line, it's not needed.

**Review by AJPfleger — DISMISSED** [review]

File: Plugins/GeoModel/include/Acts/Plugins/GeoModel/GeoModelReader.hpp [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+class GeoVPhysVol;
```
Comment (andiwand) [inline Plugins/GeoModel/include/Acts/Plugins/GeoModel/GeoModelReader.hpp:11]: unused?

**Review by andiwand — APPROVED** [review]

File: Plugins/GeoModel/include/Acts/Plugins/GeoModel/GeoModelReader.hpp (hunk 5 above)
Comment (asalzburger) [reply Plugins/GeoModel/include/Acts/Plugins/GeoModel/GeoModelReader.hpp]: Removed.

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [ccf3dc8f61db7dd21f0e0f10ccae157991d2b09f]
### :red_circle: Some tests have failed!
Please investigate the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/7430434)!

|status|job|report|
|------|---|------|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39198807)||
|:green_circle: | [test_ActsEFTrackFit](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39198806)||
|:green_circle: | [test_ActsPersistifySeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39198805)||
|:green_circle: | [test_ActsBenchmarkWithSpot](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39198804)||
|:green_circle: | [test_ActsAnalogueClustering](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39198803)||
|:green_circle: | [test_ActsWorkflowHeavyIons](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39198802)||
|:green_circle: | [test_ActsWorkflowFastTracking](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39198801)||
|:green_circle: | [test_ActsWorkflowCached](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39198800)||
|:green_circle: | [test_ActsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39198799)||
|:green_circle: | [test_ActsValidateAmbiguityResolution](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39198798)||
|:green_circle: | [test_ActsValidateResolvedTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39198797)||
|:green_circle: | [test_ActsValidateTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39198796)||
|:green_circle: | [test_ActsValidateActsCoreSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39198795)||
|:green_circle: | [test_ActsValidateActsSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39198794)||
|:green_circle: | [test_ActsValidateSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39198793)||
|:green_circle: | [test_ActsValidateOrthogonalSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39198792)||
|:green_circle: | [test_ActsValidateClusters](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39198791)||
|:green_circle: | [test_ActsPersistifyEDM](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39198790)||
|:green_circle: | [test_ActsGSFRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39198789)||
|:green_circle: | [test_ActsKfRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39198788)||
|:green_circle: | [test_ActsExtrapolationAlgTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39198787)||
|:green_circle: | [test_ActsITkTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39198786)||
|:red_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39198785)||
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39198784)||
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39198783)||
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39198782)||
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39198781)||
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39198780)||
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/39198779)||

**Pull request merged**